src/lib/route-builder.js

```diff
@@ -83,24 +84,42 @@ class RouteBuilder {
       if (traderDebit) {
         traderDebit.account = nextHop.destinationDebitAccount
       }
-      return finalTransfer
+
+      // Verify finalTransfer.credits[0].amount <= nextHop.destinationAmount
+      const finalAmount = new BigNumber(finalTransfer.credits[0].amount)
+      if (finalAmount.greaterThan(nextHop.destinationAmount)) {
+        throw new UnacceptableRateError('Payment rate does not match the rate currently offered')
+      }
+      // TODO: Verify atomic mode notaries are trusted
+      // TODO: Verify expiry is acceptable
+
+      nextHop.destinationCreditAccount = finalTransfer.credits[0].account
+      nextHop.destinationAmount = finalTransfer.credits[0].amount
+    }
+
+    const noteToSelf = {
+      source_transfer_ledger: sourceTransfer.ledger,
+      source_transfer_id: sourceTransfer.id
     }
 
     return _.omitBy({
-      id: nextHop.destinationLedger + '/transfers/' + uuid(),
+      // TODO: Use a real secret
```

Comment:
should explain what this secret is used for (this is the kind of TODO everyone forgets what it was in there for)